src/libs/blueprint/conduit_blueprint_o2mrelation_index.cpp

```diff
@@ -0,0 +1,220 @@
+// Copyright (c) Lawrence Livermore National Security, LLC and other Conduit
+// Project developers. See top-level LICENSE AND COPYRIGHT files for dates and
+// other details. No copyright assignment is required to contribute to Conduit.
+
+//-----------------------------------------------------------------------------
+///
+/// file: conduit_blueprint_o2mrelation_index.cpp
+///
+//-----------------------------------------------------------------------------
+#include <sstream>
+
+#include "conduit_blueprint_o2mrelation.hpp"
+#include "conduit_blueprint_o2mrelation_index.hpp"
+
+#include "conduit_error.hpp"
+#include "conduit_utils.hpp"
+
+//-----------------------------------------------------------------------------
+// -- begin conduit:: --
+//-----------------------------------------------------------------------------
+namespace conduit
+{
+
+//-----------------------------------------------------------------------------
+// -- begin conduit::blueprint --
+//-----------------------------------------------------------------------------
+namespace blueprint
+{
+
+//-----------------------------------------------------------------------------
+// -- begin conduit::blueprint::o2mrelation --
+//-----------------------------------------------------------------------------
+namespace o2mrelation
+{
+
+//-----------------------------------------------------------------------------
+//-----------------------------------------------------------------------------
+// O2MIndex
+//-----------------------------------------------------------------------------
+//-----------------------------------------------------------------------------
+
+//-----------------------------------------------------------------------------
+// O2MIndex Construction and Destruction
+//-----------------------------------------------------------------------------
+
+
+//---------------------------------------------------------------------------//
+O2MIndex::O2MIndex(const Node *node)
+: m_sizes_node(NULL),
+  m_indices_node(NULL),
+  m_offsets_node(NULL)
+{
+    std::vector<std::string> paths = conduit::blueprint::o2mrelation::data_paths(*node);
+
+    if(node->has_child("sizes"))
+        m_sizes_node = node->fetch_ptr("sizes");
```

Comment:
Q: Should we build the accessors in constructor, so they aren't built on the fly during the lookup calls?

Reply:
https://github.com/LLNL/conduit/pull/1080 this might help.

Reply:
Done

Reply:
Following our recent changes to NDInde - how about we avoid the node pointers here as well and only use the accessors in O2MIndex as well?